File: Session16/55. Jump Game.java
class Solution {
    public boolean canJump(int[] nums) {
        int n=nums.length;
        int lastpos=n-1;
        
        for(int i=n-2;i>=0;i--)
        {
            if(i+nums[i]>=lastpos)
            {
                lastpos=i;
            }
        }
        return lastpos==0;
    }
}

File: Session16/55. Jump Game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n=len(nums)
        lastpos=n-1
        
        for i in range(n-1,-1,-1):
            if nums[i]+i>=lastpos:
                lastpos=i
        
        return lastpos==0